python/ray/serve/autoscaling_policy.py

```diff
@@ -60,14 +60,12 @@ def _calculate_desired_num_replicas(
     smoothed_error_ratio = 1 + ((error_ratio - 1) * smoothing_factor)
     desired_num_replicas = math.ceil(num_running_replicas * smoothed_error_ratio)
 
-    # If error_ratio = 0, meaning there is no more traffic, and desired
-    # num replicas is stuck at a positive number due to the math.ceil
-    # above, decrease desired_num_replicas by one so that the deployment
-    # can eventually scale to 0.
+    # If desired num replicas is "stuck" because of the smoothing factor
+    # (meaning the traffic is low enough for the replicas to downscale
+    # without the smoothing factor), decrease desired_num_replicas by 1.
     if (
-        error_ratio == 0
+        math.ceil(num_running_replicas * error_ratio) < num_running_replicas
         and desired_num_replicas == num_running_replicas
-        and desired_num_replicas >= 1
```

Comment:
Noticed some drastically changed behaviour in the `downscaling_factor` after a Ray update and found this. I get the point, but I would argue addinf this non-linearity to the downscaling factor is much less intuitive. Previously very-low scaling factors (<0.6) were quite useful for drastic changes in day/night traffic. Now for low traffic with high bursts, we need to carefully tune the `metric_interval` and `downscale_delay`.

The previous solution allowed scaling to 0 without changing the linearity of the factor. Any chance for a reversion here? We've had to roll back to an older Ray version on our inference clusters.